src/api-service/__app__/onefuzzlib/azure/nsg.py

```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import logging
+import os
+from typing import Any, Dict, List, Optional, Union, cast
+from uuid import UUID
+
+from azure.core.exceptions import ResourceNotFoundError
+from azure.mgmt.network.models import (
+    NetworkInterface,
+    NetworkSecurityGroup,
+    SecurityRule,
+    SecurityRuleAccess,
+    SecurityRuleProtocol,
+)
+from msrestazure.azure_exceptions import CloudError
+from onefuzztypes.enums import ErrorCode
+from onefuzztypes.models import Error
+from onefuzztypes.primitives import Region
+from pydantic import BaseModel, validator
+
+from .creds import get_base_resource_group
+from .network_mgmt_client import get_network_client
+
+
+def get_nsg(name: str) -> Optional[NetworkSecurityGroup]:
+    resource_group = get_base_resource_group()
+
+    logging.debug("getting nsg: %s", name)
+    network_client = get_network_client()
+    try:
+        return cast(
+            NetworkSecurityGroup,
+            network_client.network_security_groups.get(resource_group, name),
+        )
+    except (ResourceNotFoundError, CloudError) as err:
+        logging.debug("nsg %s does not exist %s", name, err)
+        return None
+
+
+def create_nsg(name: str, location: Region) -> Union[None, Error]:
```

Comment:
'name' is the region, right? Is this appropriate? It's not really the NSG for the region, it's the NSG for the proxy _for a particular region_. As some people mentioned in reviewing my PR, we might leave room for future NSGs which are more general than proxy. 

I'm just proposing that we maybe name the NSG something like 'proxy-nsg-GUID' or something of that nature.

Reply:
I guess I was making a more general point and using this function as a place to make it. 

The question is - 'do we want to name the NSG the region?' Referring to your point above, if there are multiple NSGs per region, the this doesn't make sense.

Reply:
From NSG class point of view - it is not supposed to know how we are naming NSGs. 
I am assigning NSG name to be region since it is one resource per region. Exactly in the same manner as VNETs. They are named same as the region (but in the code they accept name and region separately)